Namespace reconcile

bskaplan opened the pull request:
Fixes #2397 

## Proposed Changes

- Switch Namespace reconciliation from using a Tracker to the cache for watching owned objects

**Release Note**

<!-- Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". -->

```release-note
Changes the way namespaces are reconciled to match other resources. This may result in existing brokers and their service accounts not reconciling properly after this update is applied. 
```

matzew commented:
/ok-to-test

vaikas commented:
Thanks for doing this! After these changes, looks like there's only one use of Tracker? Using Broker? If that's the only remaining use of tracker, should we just change that as well?

bskaplan commented:
Whoops, missed one of the uses. Should be all fixed now.

n3wscott commented:
Have you tested this in a cluster?

from looking that the code, it looks correct assuming the owners are setup correctly from all the resources the namespace is suppose to create. But I do not see that code anywhere in the namespace reconciler. I suspect the reason the current code is using the tracker is they wanted to avoid setting the namespace as the owner/controller of the broker and service account resources.

Without setting an owners, `Handler:    controller.HandleAll(impl.EnqueueControllerOf),` will do nothing and I think the end result of this PR will be a reconciler that creates the resources but never sees changes to those resources until the resync period. But this is just from reading the code and not installing this patch in a cluster, so please confirm.

bskaplan commented:
Updated release notes, and also updated the PR to also stop using the tracker for ConfigMapPropagation, which was added after I started working on this.

grantr commented:
/hold cancel
